File: Exercism/javascript/phone-number/phone-number.js
'use strict'

export class PhoneNumber {
  constructor (number) {
    this.nr = number.trim()
  }

  number () {
    const NANP = /^(?:\+?1[-. ]?)?\(?([2-9][0-9]{2})\)?(?:[-. ]+)?([2-9][0-9]{2})(?:[-. ]+)?([0-9]{4})$/
    if (NANP.test(this.nr)) {
      return NANP.exec(this.nr).splice(1).join('')
    }
    return null
  }
}

File: Exercism/python/phone-number/phone_number.py
import re


class PhoneNumber:
    def __init__(self, number: str) -> object:
        """Stores a phonenumber according NANP"""
        number = re.sub(" +", " ", number).strip()
        nr = re.match((r"^(?:\+?1)?\W?"
                       r"\(?([2-9][0-9]{2})\)?\W?"
                       r"([2-9][0-9]{2})\W?"
                       r"([0-9]{4})$"), number)
        if nr is None:
            raise ValueError("Not a phonenumber")
        self.number = ''.join(nr.groups())
        self.area_code = nr.group(1)

    def pretty(self) -> str:
        """Pretify phonenumber"""
        return f"({self.area_code}) {self.number[3:6]}-{self.number[6:]}"
